p2p/pex/addrbook.go

```diff
@@ -605,19 +609,25 @@ func (a *addrBook) pickOldest(bucketType byte, bucketIdx int) *knownAddress {
 // adds the address to a "new" bucket. if its already in one,
 // it only adds it probabilistically
 func (a *addrBook) addAddress(addr, src *p2p.NetAddress) error {
+	if addr == nil {
```

Comment:
why not
```go
if addr == nil || src == nil {
```
since we return the exact same ordering of arguments?

Reply:
originally I used different errors, then consolidated them. thanks